src/api-utils.js

```diff
@@ -56,17 +60,24 @@ function buildFetchRequest(
   }
 
   if (formData) {
-    options.body = toUrlParams(formData);
-    if (!caseInsensitiveHasKey(headers, "Content-Type")) {
-      options.headers["Content-Type"] = "application/x-www-form-urlencoded";
+    let transformData = identity;
+
+    if (!isFormDataObj(formData)) {
+      transformData = toUrlParams;
+
+      if (!caseInsensitiveHasKey(headers, "Content-Type")) {
+        options.headers["Content-Type"] = "application/x-www-form-urlencoded";
+      }
     }
+
+    options.body = transformData(formData);
```

Comment:
It seems like transformData is only used here, Maybe we could do `options.body = toUrlParams(formData)` into the `if` statement and remove the `transformData` function.